lte/gateway/python/magma/mobilityd/ip_descriptor_map.py

```diff
@@ -48,7 +48,8 @@ class IpDescriptorMap:
 
     def __init__(self,
```

Comment:
I'd like us to get rid of the complexity in the constructor and just dependency inject the backing map. We should instantiate the backing map, whether it's a defaultdict(dict) or a redis-backed store, at the caller. Then we can decouple this storage interface from the operational mode of mobilityd (stateless or in-memory)

Reply:
I think maybe a `MutableMapping[MutableMapping[str, IPDesc]]` would be the right type annotation for the injected `ip_states`

Reply:
I believe we have two options here (In the current version I followed the second one):

1. Dependency inject just the backing-map infrastructure:
    - Create an object with the redis or memory-map backend
    - Inject this object into `IPAddressManager` and let it select which IPAllocation method (static/dhcp) will be used.
    - In this one the creation of IPAllocator is still responsibility of `IPAddressManager`, though the backend is transparent.
2. Dependency inject the whole `IPAllocator` and auxiliary objects.
    - Create the expected `IPAllocator` in the beginning, already configured for static/dhcp with redis/in-memory backend.
    - `IPAddressManager` receives it and use as usual.

I followed the second one, since it appears to be more flexible.